tests/cmdline/commands/test_process.py

```diff
@@ -549,23 +576,157 @@ def test_process_kill(submit_and_await, run_cli_command):
+
+    run_cli_command(cmd_process.process_kill, [str(node.pk), '-F', '--wait'])
     await_condition(lambda: node.is_killed)
+    assert node.process_status == 'Force killed through `verdi process kill -F`'
+
+    # 3) Kill a running process
+    node = submit_and_await(make_a_builder(sleep_seconds=100), ProcessState.RUNNING)
+    run_cli_command(cmd_process.process_kill, [str(node.pk), '--wait'])
+
+    # note: timeout should be less than sleep_seconds for this test to be valid
+    # this is to make sure that the process is still running before we kill it
+    await_condition(lambda: node.is_killed, timeout=kill_timeout)
     assert node.process_status == 'Killed through `verdi process kill`'
 
+    # 4) *Force* kill a running process
+    node = submit_and_await(make_a_builder(sleep_seconds=100), ProcessState.RUNNING)
+    result = run_cli_command(cmd_process.process_kill, [str(node.pk), '-F', '--wait'])
+
+    await_condition(lambda: node.is_killed, timeout=kill_timeout)
+    assert node.process_status == 'Force killed through `verdi process kill -F`'
+
+    # 5) `verdi process kill --all` should kill all processes (running / not running)
+    node_1 = submit_and_await(WaitProcess, ProcessState.WAITING)
+    run_cli_command(cmd_process.process_pause, [str(node_1.pk), '--wait'])
+    await_condition(lambda: node_1.paused)
+    node_2 = submit_and_await(make_a_builder(sleep_seconds=100), ProcessState.RUNNING)
+
+    run_cli_command(cmd_process.process_kill, ['--all', '--wait'], user_input='y')
+    await_condition(lambda: node_1.is_killed, timeout=kill_timeout)
+    await_condition(lambda: node_2.is_killed, timeout=kill_timeout)
+    assert node_1.process_status == 'Killed through `verdi process kill`'
+    assert node_2.process_status == 'Killed through `verdi process kill`'
+
+    # 6) `verdi process kill --all -F` should Force kill all processes (running / not running)
+    node_1 = submit_and_await(WaitProcess, ProcessState.WAITING)
+    run_cli_command(cmd_process.process_pause, [str(node_1.pk), '--wait'])
+    await_condition(lambda: node_1.paused)
+    node_2 = submit_and_await(make_a_builder(sleep_seconds=100), ProcessState.RUNNING)
+
+    run_cli_command(cmd_process.process_kill, ['--all', '--wait', '-F'], user_input='y')
+    await_condition(lambda: node_1.is_killed, timeout=kill_timeout)
+    await_condition(lambda: node_2.is_killed, timeout=kill_timeout)
+    assert node_1.process_status == 'Force killed through `verdi process kill -F`'
+    assert node_2.process_status == 'Force killed through `verdi process kill -F`'
+
+    # 7) *Force* kill a process that has stuck in EBM, something that *kill* cannot do.
+    # `verdi process kill -F` --as the first attempt--
+    def mock_open(self):
+        raise Exception('Mock open exception')
+
+    # patch a faulty transport open, to make EBM go crazy
+    inject_patch.patch('aiida.transports.plugins.local.LocalTransport.open', mock_open)
```

Comment:
I think the way to implement an inject_patch fixture is quite hacky. Why not defined a MockTransport with `open` method implemented as mock_open?

Reply:
Well, the main problem I had here was, the daemon being instantiated from another run of the interpreter so it couldn't see any mocking function you defined here.. So that I made this hard patch fixture, which might not be an ideal way, as we discussed before.

Anyways, I remember @agoscinski had a suggestion that maybe he could instantiate the daemon on the same running interpreter so what could just mock those functions normally. 

@agoscinski can you try your idea, to see if that works?